tests/integ/test_udf.py

```diff
@@ -1185,12 +1185,10 @@ def is_numpy_available() -> bool:
     # add module objects
     # but we can't register a udf with these versions
     # because the server might not have them
-    session.add_packages(numpy, pandas, dateutil)
-    assert session.get_packages() == {
-        "numpy": f"numpy=={numpy.__version__}",
-        "pandas": f"pandas=={pandas.__version__}",
-        "python-dateutil": f"python-dateutil=={dateutil.__version__}",
-    }
```

Comment:
This part uses the public API `session.get_packages`. From test perspective, it's different from testing `session._resolve_packages`. I suggest you keep this part and add new tests for `session._resolve_packages`

Reply:
So there is a behavior change here. We need to check the behavior change process.